### PR TITLE
CS Falcon - XSUP 38920

### PR DESCRIPTION
**Pull request by RotemAmit**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-38920)

## Description
* Added debug logs to better monitor the response from CS Falcon API calls.
* Fixed an issue where a failure in a request for a new token will result in a generic error instead of the specific error, and added a retry mechanism in case it is needed.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by MLainer1 — COMMENTED** [review]
😎

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py [hunk 1]
```diff
@@ -1420,8 +1436,9 @@ def get_token_request():
     headers = {
         'Content-Type': 'application/x-www-form-urlencoded'
     }
-    token_res = http_request('POST', '/oauth2/token', data=body, headers=headers, safe=True,
-                             get_token_flag=False)
+    demisto.debug('In get_token_request')
```
Comment (Shellyber) [inline Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py:1439]: Do we need this log?

**Review by Shellyber — APPROVED** [review]
Nice work!

File: Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py (hunk 1 above)
Comment (RotemAmit) [reply Packs/CrowdStrikeFalcon/Integrations/CrowdStrikeFalcon/CrowdStrikeFalcon.py]: deleted

**Comment by RotemAmit** [comment]
The test playbook `Test Playbook - CrowdStrike Falcon - Retrieve File` fails on XSOAR SAAS since one of the hosts is down, I opened a [ticket](https://jira-dc.paloaltonetworks.com/browse/CRTX-122151) to the lab to fix it.
The test playbook failed only on XSOAR SAAS and not XSOAR 6 because a Playback was used.

**Pull request merged**